src/catalogue/category/cataloguePropertiesForm.component.tsx

```diff
@@ -390,25 +393,111 @@ function CataloguePropertiesForm(props: CataloguePropertiesFormProps) {
+              }
+              value={
+                catalogueItemPropertyTemplates?.find(
+                  (template) => template.name === field.name
+                ) || null
+              }
+              inputValue={field.name}
+              onInputChange={(_event, newValue) => {
+                handleChange(field.cip_placement_id, 'name', newValue);
+              }}
+              onChange={(_event, newValue) => {
+                if (typeof newValue !== 'string' && newValue !== null) {
+                  const updatedFormFields = [...formFields];
+                  const fieldIndex = updatedFormFields.findIndex(
+                    (formField) =>
+                      formField.cip_placement_id === field.cip_placement_id
+                  );
+                  if (fieldIndex === -1) {
+                    return;
+                  }
+
+                  // If the field has allowed values, assign an av_placement_id to each value
+                  if (
+                    newValue.allowed_values &&
+                    newValue.allowed_values.values
+                  ) {
+                    const updatedAllowedValues =
+                      newValue.allowed_values.values.map((value) => ({
+                        av_placement_id: generateUniqueId('av_placement_id_'),
+                        value: value,
+                      }));
+
+                    newValue = {
+                      ...newValue,
+                      allowed_values: {
+                        ...newValue.allowed_values,
+                        values: updatedAllowedValues,
+                      },
+                    };
+                  }
+                  const { id, ...newValueWithoutId } = newValue;
+                  updatedFormFields[fieldIndex] = {
+                    ...updatedFormFields[fieldIndex],
+                    ...newValueWithoutId,
+                  };
+                  onChangeFormFields(updatedFormFields);
+
+                  const updatedCatalogueItemPropertiesErrors =
+                    catalogueItemPropertiesErrors.filter(
+                      (item) => item.cip_placement_id !== field.cip_placement_id
+                    );
+
+                  onChangeCatalogueItemPropertiesErrors(
+                    updatedCatalogueItemPropertiesErrors.filter(
+                      (item) =>
+                        item.errors?.errorMessage !==
+                        'Duplicate property name. Please change the name or remove the property'
+                    )
+                  );
+
+                  const updatedAllowedValuesListErrors =
+                    allowedValuesListErrors.filter(
+                      (item) => item.cip_placement_id !== field.cip_placement_id
+                    );
+
+                  onChangeAllowedValuesListErrors(
+                    updatedAllowedValuesListErrors
+                  );
+                }
+              }}
```

Comment:
I think it would be good to apply you're suggestion to avoid repeating this logic.

Reply:
I don't think what I suggested yesterday is going to work well. The functions in this file currently work well for adding, deleting, updating one field at a time and this onChange updates many fields at time.

Reply:
Ok, and its also not sensible to put it into a function to do one update, and then just repeat it for each field for the onChange?

Reply:
That's over complicates things. This onChange is simple at add all and clear all errors

Reply:
to use the current functions in this file you would have to use the handleAdd and then handleChange. This problem because more problematic for the allowed values (loop through a long list doing handleAdd and handleChange)